listener/src/main.rs

```diff
@@ -0,0 +1,232 @@
+    static ref BLOCK_POLLING_INTERVAL: Duration = Duration::from_millis(1000);
+}
+
+fn main() {
+    use std::sync::Mutex;
+    use tokio::runtime;
+
+    // Create components for tokio context: multi-threaded runtime, executor
+    // context on the runtime, and Timer handle.
+    //
+    // Configure the runtime to shutdown after a panic.
+    let runtime: Arc<Mutex<Option<runtime::Runtime>>> = Arc::new(Mutex::new(None));
+    let handler_runtime = runtime.clone();
+    *runtime.lock().unwrap() = Some(
+        runtime::Builder::new()
+            .core_threads(100)
+            .panic_handler(move |_| {
+                let runtime = handler_runtime.clone();
+                std::thread::spawn(move || {
+                    if let Some(runtime) = runtime.lock().unwrap().take() {
+                        // Try to cleanly shutdown the runtime, but
+                        // unconditionally exit after a while.
+                        std::thread::spawn(|| {
+                            std::thread::sleep(Duration::from_millis(3000));
+                            std::process::exit(1);
+                        });
+                        runtime
+                            .shutdown_now()
+                            .wait()
+                            .expect("Failed to shutdown Tokio Runtime");
+                        println!("Runtime cleaned up and shutdown successfully");
+                    }
+                });
+            })
+            .build()
+            .unwrap(),
+    );
+
+    let mut executor = runtime.lock().unwrap().as_ref().unwrap().executor();
+    let mut enter = tokio_executor::enter()
+        .expect("Failed to enter runtime executor, multiple executors at once");
+    let timer = Timer::default();
+    let timer_handle = timer.handle();
+
+    // Setup runtime context with defaults and run the main application
+    tokio_executor::with_default(&mut executor, &mut enter, |enter| {
+        tokio_timer::with_default(&timer_handle, enter, |enter| {
+            enter
+                .block_on(future::lazy(async_main))
+                .expect("Failed to run main function");
+        })
+    });
+}
+
+fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
+    env_logger::init();
+
+    let postgres_url = env::var("POSTGRES_URL").expect("Specify POSTGRES_URL variable");
+    let ethereum_node_url = env::var("ETHEREUM_NODE_URL").expect("Specify ETHEREUM_RPC variable");
+    let network_name = env::var("NETWORK_NAME").expect("Specify NETWORK_NAME variable");
+
+    let http_port = env::var("GRAPHQL_PORT")
+        .expect("Specify GRAPHQL_PORT variable")
+        .parse::<u16>()
+        .expect("Couldn't parse GRAPHQL_PORT variable as u16");
+    let ws_port = env::var("WS_PORT")
+        .expect("Specify WS_PORT variable")
+        .parse::<u16>()
+        .expect("Couldn't parse WS_PORT variable as u16");
+
+    let logger = logger(false);
```

Comment:
hopefully having the logger declared below the `log!` macro is used doesn't cause any problems.